lib/libvfio-user.c

```diff
@@ -1091,6 +1107,8 @@ vfu_reset_ctx(vfu_ctx_t *vfu_ctx, const char *reason)
                                           vfu_ctx);
     }
 
+    handle_device_reset(vfu_ctx, VFU_RESET_LOST_CONN);
```

Comment:
Hmm, I'm not really a fan of handle_*() being called outside of the process_request() path. Can we keep handle_device_reset() for the message, and move the handlers down into a function that handle_device_reset() itself calls, as well as here?

Reply:
Sure.